### PR TITLE
Support bullet decals

### PR DESCRIPTION
**Pull request by Frickerson**

### CODE REVIEW

File: Source/ActorPool/Private/ActorPoolSettings.cpp [hunk 1]
```diff
@@ -1,15 +1,18 @@
...
     bEnableCollision( true ),
     bDisableNetDormancy( true ),
     NetDormancy( ENetDormancy::DORM_Awake )
 {}
 
-FActorPoolInfos::FActorPoolInfos():
+FActorPoolInfos::FActorPoolInfos() :
     Count( 0 ),
-    bAllowNewInstancesWhenPoolIsEmpty( true )
+    bAllowNewInstancesWhenPoolIsEmpty( true ),
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolSettings.cpp]: there's no point in having one bool to allow to create new instances and one other to use a circular buffer. It's either one of the other
I would use a policy enum

File: Source/ActorPool/Private/ActorPoolActor.cpp [hunk 2]
```diff
@@ -192,7 +201,10 @@ void AActorPoolActor::BeginPlay()
     {
         for ( const auto & pool_infos : settings->PoolInfos )
         {
-            ActorPools.Emplace( pool_infos.ActorClass.LoadSynchronous(), CreateActorPoolInstance( pool_infos ) );
+            if ( HasAuthority() && pool_infos.bSpawnOnServer || !HasAuthority() && pool_infos.bSpawnOnClients )
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolActor.cpp]: Proper check would be something like:

```
const auto is_standalone = UKismetSystemLibrary::IsStandalone( world );
    auto is_server = IsRunningDedicatedServer();

#if WITH_EDITOR
    checkSlow( game_instance->GetWorldContext() );
    is_server |= world->GetGameInstance()->GetWorldContext()->RunAsDedicated;
#endif

    const auto is_client = !is_server;

    if ( is_server && pool_infos.bSpawnOnServer || is_client && pool_infos.bSpawnOnClients )
```

Check `UGFEGameFeatureAction_SpawnActors`

File: Source/ActorPool/Private/ActorPoolActor.cpp [hunk 3]
```diff
@@ -169,17 +182,13 @@ AActor * FActorPoolInstances::SpawnActorAndAddToInstances( UWorld * world )
 AActorPoolActor::AActorPoolActor()
 {
     PrimaryActorTick.bCanEverTick = false;
+    bReplicates = true;
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolActor.cpp]: why is this needed?

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/ActorPool/Private/ActorPoolActor.cpp (hunk 3 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolActor.cpp]: Removed.

File: Source/ActorPool/Private/ActorPoolActor.cpp (hunk 2 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolActor.cpp]: Done.

File: Source/ActorPool/Private/ActorPoolSettings.cpp (hunk 1 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolSettings.cpp]: Done.

File: Source/ActorPool/Private/ActorPoolActor.cpp [hunk 4]
```diff
@@ -2,13 +2,15 @@
 
 #include "APPooledActorInterface.h"
 #include "ActorPoolSubSystem.h"
+#include "Engine/Engine.h"
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolActor.cpp]: <>

File: Source/ActorPool/Private/ActorPoolActor.cpp [hunk 5]
```diff
@@ -2,13 +2,15 @@
 
 #include "APPooledActorInterface.h"
 #include "ActorPoolSubSystem.h"
+#include "Engine/Engine.h"
+#include "Kismet/KismetSystemLibrary.h"
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolActor.cpp]: same

File: Source/ActorPool/Private/ActorPoolActor.cpp [hunk 6]
```diff
@@ -55,6 +57,14 @@ AActor * FActorPoolInstances::GetAvailableInstance( UWorld * world )
         {
             SpawnActorAndAddToInstances( world );
         }
+        else if ( PoolInfos.PoolingPolicy == EAPPoolingPolicy::LoopInstances )
```
Comment (TheEmidee) [inline Source/ActorPool/Private/ActorPoolActor.cpp]: use a switch

File: Source/ActorPool/Public/ActorPoolSettings.h [hunk 7]
```diff
@@ -2,13 +2,21 @@
...
+#include <Templates/SubclassOf.h>
 
 #include "ActorPoolSettings.generated.h"
 
 class AActor;
 
+UENUM()
+enum class EAPPoolingPolicy : uint8
+{
+    None,
```
Comment (TheEmidee) [inline Source/ActorPool/Public/ActorPoolSettings.h]: can't be none

**Review by TheEmidee — CHANGES_REQUESTED** [review]

File: Source/ActorPool/Private/ActorPoolActor.cpp (hunk 6 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolActor.cpp]: Done.

File: Source/ActorPool/Public/ActorPoolSettings.h (hunk 7 above)
Comment (Frickerson) [reply Source/ActorPool/Public/ActorPoolSettings.h]: Done.

File: Source/ActorPool/Private/ActorPoolActor.cpp (hunk 5 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolActor.cpp]: Done.

File: Source/ActorPool/Private/ActorPoolActor.cpp (hunk 4 above)
Comment (Frickerson) [reply Source/ActorPool/Private/ActorPoolActor.cpp]: Done.

**Review by TheEmidee — APPROVED** [review]
